scripts/extractDtsModules.js

```diff
@@ -0,0 +1,57 @@
+const tmp = require("tmp");
+const fs = require("fs-extra");
+const _ = require("lodash");
+const { DiagnosticCategory } = require("typescript");
+
+const createTsProgram = require("./createTypescriptProgram");
+
+tmp.setGracefulCleanup(); // cleans tmp file on process exit
+
+/**
+ * Get a typescript compiled program and if has compile time error will throw error with
+ * a formalized errors breakdown
+ * @param {typescript.Program} program
+ */
+function detectErrors(program) {
+  const syntaxErrors = program
+    .getSyntacticDiagnostics()
+    .filter(diagnostic => diagnostic.category === DiagnosticCategory.Error);
```

Comment:
Maybe we should move this into `createTypeScriptProgram.js`?
That way no matter what the use case if the program has errors we'll file the process.

Reply:
do we need `detectErrors` code? Isn’t it the current situation? if for some reason there is an error in one of the build scripts the whole build process is falling? did you write it for better debugging?

Reply:
Surprisingly no!
I was thought the complier will error out, but it just reports it this way.
The worst thing is that the other results give out some wired shit. Like for example the bad file is missing a ' in the model name so the return result for the module name `hello-world { \n`

Reply:
I went over the changes after I pushed and figured that could happen in the full-types generation as well and we can and up pushing bad d.ts files to the client.
Question is does throwing an error in that case is the right way to go

Reply:
Ok so I made decision to move deeper to effect the other scripts as well.
I think it is for the best